src/server/page.ts

```diff
@@ -378,7 +378,9 @@ export class Page extends EventEmitter {
     if (this._closedState !== 'closing') {
       this._closedState = 'closing';
       assert(!this._disconnected, 'Protocol error: Connection closed. Most likely the page has been closed.');
-      await this._delegate.closePage(runBeforeUnload);
+      // This might throw if the browser context containing the page closes
+      // while we are trying to close the page.
+      await this._delegate.closePage(runBeforeUnload).catch(e => {});
```

Comment:
From what I can see, we also handle stopping screencast in `closePage`. I am worried that if something throws while closing, we won't stop/finish screencast and `context.close()` will wait for that forever.

Reply:
The only expected errors in closePage actually result in the page being closed. In chromium, the close event will fire on the page and stop the screencast. In wk/ff, the screencast is handled at the browser level so I think it should be fine?

I'd like to have an assertion that the page was closed inside that catch, but I think in chromium the close event could come later.

Reply:
Hmm... Can we log the error, say to `pw:error`, so that later on while debugging the impossible bug, we would know where to look at?